source/common/tracing/http_tracer_impl.cc

```diff
@@ -168,11 +168,10 @@ HttpTracerImpl::HttpTracerImpl(DriverPtr&& driver, const LocalInfo::LocalInfo& l
 
 SpanPtr HttpTracerImpl::startSpan(const Config& config, Http::HeaderMap& request_headers,
                                   const Http::AccessLog::RequestInfo& request_info) {
-  std::string span_name = HttpTracerUtility::toString(config.operationName());
+  std::string span_name;
```

Comment:
what is the span name we are going to report for the ingress?

Reply:
For zipkin tracer, this was not set - so only change is that the Ingress value would not be provided to LightStep's tracer - so before this PR is merged this needs to be checked with them.

Reply:
@bhs Could you provide some guidance here?

Prior to this PR, this code was building a `span_name` based on the 'OperationName` (misleading) which essentially just represents the direction of the request (Ingress or Egress), and if Egress, then it appends the host. However the Zipkin tracer code was ignoring this value, and instead using the host for Egress requests and empty span name for Ingress.

It appears that the `span_name` is just being passed through to the LightStep tracer and not doing any kind of interpretation of the value to determine span kind. So assume this value is just being included in the LightStep spans and displayed via your UI?

If so, do you have any objections to the changed values (i.e. blank on Ingress and host for Egress), or would you prefer different default values? With this PR, if the configuration includes a specified `operation` for the route, then this would be used instread.

Reply:
LS does not care about `operation name`, but if affects `Saved Searches` if there are any. Do we really need to change this behavior?
It seems like if there is a decorator on a route, span name will be overwritten and if not, that this default one is going to be used.

Reply:
That is the problem - the default behaviour is different between the two tracers. LS uses the default value passed in, but the Zipkin tracer overrides the supplied value and just uses the Host for Egress routes.

So not sure how to allow both tracers to maintain their current default behaviour, while at the same time enabling the Decorator to override the span name in both.

Reply:
Got it, if there are no common names for spans for LS and Zipkin spans, I'd push the logic of assigning specific span names to the specific tracer (ls and zipkin tracers).

Reply:
IMO the LS current behavior makes more sense. Meaning, if no operation is set, base operation name off of egress/ingress + host, etc. If the tracer wants to override (as you do inside Zipkin) that seems fine? It doesn't make much sense to me to push this logic entirely into each tracer, but I suppose we could do that. Maybe I'm not fully following the issue here.

Reply:
@mattklein123 @RomanDzhabarov You are correct - I was still thinking about the previous way I had implemented it, where the tracing config provided operation was passed in when the span was started - whereas now it is set after the span has been created within the Decorator - so all good, I can return the default behaviour to previous versions.